### PR TITLE
Pin kubetest2 version

### PR DESCRIPTION
**Pull request by mattcary**
/kind failing-test

Pin kubetest2 version, as recent changes break the version of ginkgo we're using.

/assign @leiyiz

### CODE REVIEW

File: test/run-k8s-integration-ci.sh [hunk 1]
```diff
@@ -54,10 +54,11 @@ export GCE_PD_VERBOSITY=9
 make -C "${PKGDIR}" test-k8s-integration
 
 if [ "$use_kubetest2" = true ]; then
-    go install sigs.k8s.io/kubetest2@latest;
-    go install sigs.k8s.io/kubetest2/kubetest2-gce@latest;
-    go install sigs.k8s.io/kubetest2/kubetest2-gke@latest;
-    go install sigs.k8s.io/kubetest2/kubetest2-tester-ginkgo@latest;
+    kt2_version=0e09086b60c122e1084edd2368d3d27fe36f384f
```
Comment (leiyiz) [inline test/run-k8s-integration-ci.sh:57]: consider having a comment about where/how to update this when needed

Comment (mattcary) [reply test/run-k8s-integration-ci.sh:57]: I don't think we actually know what the process is to update this is unfortunately?

I think we should also consider that we're doing with release-tools/, as we pull that in but don't actually use it at all.

Comment (leiyiz) [reply test/run-k8s-integration-ci.sh:57]: on filestore side we said something like "ask engprod oncall what version are they using internally if we need to update"

**Pull request merged**